walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```
@@ -15,9 +15,225 @@
+    }
+
+    // both mandatory and optional attendees requested for the meeting
+    Set<String> allAttendees = new HashSet<>(mandatoryAttendees);
+    allAttendees.addAll(request.getOptionalAttendees());
+
+    // events that only include attendees requested for the meeting
+    List<Event> filteredEvents = 
+        filterEventsByAttendees(allAttendees, events);
+
+    // filtered event lists sorted by start and end times
+    List<Event> startSortedEvents = 
+        sortEvents(filteredEvents, Event.ORDER_BY_START);
+    List<Event> endSortedEvents =
+        sortEvents(filteredEvents, Event.ORDER_BY_END);
+
+    // min duration of the available time slot
+    long minDuration = request.getDuration();
+
+    // index of the current position in startSortedEvents and endSortedEvents
+    int startIndex = 0;
+    int endIndex = 0;
+
+    // The maximum value startIndex and endIndex can have 
+    int maxIndex = filteredEvents.size();
+
+    // keep track of how many mandatory and optional attendees are busy
+    int busyMandatoryAttendees = 0;
+    int busyOptionalAttendees = 0;
+
+    // keep track of when a possible available slot for 
+    // mandatory attendees and all attendees begins
+    int freeMandatorySlotStart = TimeRange.START_OF_DAY;
+    int freeOptionalSlotStart = TimeRange.START_OF_DAY;
+
+    // temp variables to access necessary events and event times
+    Event startEvent;
+    Event endEvent;
+    int startEventTime;
+    int endEventTime;
+
+    // temp variables to access everyone attending an event and 
+    // how many of those are mandatory for the meeting
+    Set<String> eventAttendees;
+    int mandatoryEventAttendees;
+
+    // go through day by iterating through times of interest
+    // (when an event ends or starts)
+    while (endIndex < maxIndex) {
+      if (busyMandatoryAttendees < 0 || busyOptionalAttendees < 0) {
+        System.out.println("!!!!CANNOT HAVE NEGATIVE ATTENDEES!!!!!");
```

If this is unreachable code (the condition always evaluates to false), a better approach would be to have an assertion so that you can have better test coverage. You can do that using `assert busyMandatoryAttendees >= 0 && busyOptionalAttendees >= 0`.

If this code is reachable due to bad input parameters, I would throw an exception. This would ensure that no other code gets executed and makes debugging the issue easier. You can use Java's RuntimeException class. Ex: `throw new RuntimeException("<error message>");`